lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb

```diff
@@ -104,6 +79,49 @@ class Console::CommandDispatcher::Stdapi::Sys
     "-c" => [ false, "Continues suspending or resuming even if an error is encountered"],
     "-r" => [ false, "Resumes the target processes instead of suspending"	   ])
 
+  #
+  # Options used by the 'shell' command.
+  #
+  def shell_opts
+    default_shell_opts = {
+      '-h' => [false, 'Help menu.'],
+      '-l' => [false, 'List available shells (/etc/shells).'],
+      '-t' => [true, 'Spawn a PTY shell (/bin/bash if no argument given).']
+    }
+    if Msf::FeatureManager.instance.enabled?(Msf::FeatureManager::FULLY_INTERACTIVE_SHELLS)
+      Rex::Parser::Arguments.new(default_shell_opts.merge(
+        { '-i' => [ false, 'Drop into a fully interactive shell. (Only used in conjunction with `-t`'] }
+      ))
+    else
+      Rex::Parser::Arguments.new(default_shell_opts)
+    end
+  end
```

Comment:
```suggestion
  def shell_opts
    if framework.features.enabled?(Msf::FeatureManager::FULLY_INTERACTIVE_SHELLS)
      @@shell_opts_with_fully_interactive_shell
    else
      @shell_opts
    end
  end
```

Reply:
The instance of the feature manager is available on `framework`

Reply:
yea tried that before and I thought it didn;t work, so I replicated it just now
```meterpreter > shell -it
[-] Error running command shell: NameError undefined local variable or method `framework' for #<Rex::Post::Meterpreter::Ui::Console::CommandDispatcher::Stdapi::Sys:0x00007ff6e587a1c0>
```

Reply:
Ah, for meterpreter's context it would be `client.framework.features.enabled?(...)`

Reply:
aye that's worked, cheers